internal/operands/metrics/resources.go

```diff
@@ -61,31 +72,87 @@ func ServiceMonitorLabels() map[string]string {
 	}
 }
 
-func newServiceMonitorCR(namespace string) *promv1.ServiceMonitor {
-	return &promv1.ServiceMonitor{
+func serviceCABundle() promv1.SecretOrConfigMap {
+	return promv1.SecretOrConfigMap{
+		ConfigMap: &v1.ConfigMapKeySelector{
+			LocalObjectReference: v1.LocalObjectReference{
+				Name: ServiceCABundle,
+			},
+			Key: ServiceCABUndleKey,
+		},
+	}
+}
+
+func olmManagedCABundle() promv1.SecretOrConfigMap {
+	return promv1.SecretOrConfigMap{
+		Secret: &v1.SecretKeySelector{
```

Comment:
Please add a comment, that OLM uses a self-signed CA, and that it is not available in a ConfigMap.
So that future developers know why we use a secret.